### PR TITLE
PWN-3886 - Fix renBTC receiving, temporary add devnet for test

### PR DESCRIPTION
**Pull request by davran1001**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-3886

## What was done

Fix minting renBTC

### CODE REVIEW

**Review by kabramovp2p — APPROVED** [review]

**Review by kuznetsov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/infrastructure/network/environment/NetworkEnvironmentManager.kt [hunk 1]
```diff
@@ -27,7 +27,8 @@ class NetworkEnvironmentManager(
     private fun loadAvailableEnvironments(): List<NetworkEnvironment> {
         val networksFromRemoteConfig = networkListFeatureToggle.value.map { it.url }
         val isNetworkAvailable = { network: NetworkEnvironment -> network.endpoint in networksFromRemoteConfig }
-        return NetworkEnvironment.values().filter(isNetworkAvailable)
+        //Temporary remove for test
```
[the remark this answers is not in the document]
Comment (davran1001) [reply app/src/main/java/org/p2p/wallet/infrastructure/network/environment/NetworkEnvironmentManager.kt]: Yes,  after PR will be approved, i will remove this

File: app/src/main/java/org/p2p/wallet/renbtc/service/RenStatusExecutor.kt [hunk 2]
```diff
@@ -12,7 +12,7 @@ import org.p2p.wallet.receive.renbtc.BTC_DECIMALS
 import org.p2p.wallet.utils.fromLamports
 import timber.log.Timber
 
-private const val ONE_MINUTE_DELAY = 1000L * 60L
+private const val ONE_MINUTE_DELAY = 1000L * 30L
```
[the remark this answers is not in the document]
Comment (davran1001) [reply app/src/main/java/org/p2p/wallet/renbtc/service/RenStatusExecutor.kt:15]: and this too, changed it for more accurate test

File: app/src/main/java/org/p2p/wallet/renbtc/service/RenVMService.kt [hunk 3]
```diff
@@ -69,28 +76,47 @@ class RenVMService : Service(), CoroutineScope {
...
+     * This will start new session anyway
+     * This can be called when user clicked [show address] and wants to make transaction
+     * */
+    private fun startNewSession() {
+        if (renVMJob?.isActive == true) {
+            Timber.tag(TAG).d("Service is already active, skipping new session generate")
+            return
+        }
+
+        Timber.tag(TAG).d("Generating new session")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/renbtc/service/RenVMService.kt:110]: maybe we should move from .d to .i for better logs in crashes?

File: solanaj/src/main/java/org/p2p/solanaj/kits/renBridge/GatewayStateData.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package org.p2p.solanaj.kits.renBridge
+
+import org.p2p.solanaj.core.AbstractData
+import org.p2p.solanaj.utils.ByteUtils
+import java.math.BigInteger
+
+private const val GATEWAY_STATE_DATA_LENGTH = 1 + 20 + 32 + ByteUtils.UINT_64_LENGTH + 1
```
Comment (gslevinkov) [inline solanaj/src/main/java/org/p2p/solanaj/kits/renBridge/GatewayStateData.kt:7]: need a break line here

File: solanaj/src/main/java/org/p2p/solanaj/kits/renBridge/GatewayStateData.kt [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+class GatewayStateData(private val data: ByteArray) : AbstractData(data, GATEWAY_STATE_DATA_LENGTH) {
+
+    val isInitialized: Boolean = readByte().toInt() != 0
+    val renVMAuthority: ByteArray = readBytes(20)
+    private val selectors: ByteArray = readBytes(32)
+    val burnCount: BigInteger = readUint64()
+    private val underlyingDecimals: Byte = readByte()
+
+    companion object {
+        fun decode(data: ByteArray) = GatewayStateData(data)
```
Comment (gslevinkov) [inline solanaj/src/main/java/org/p2p/solanaj/kits/renBridge/GatewayStateData.kt:17]:
```suggestion
        fun decode(data: ByteArray): GatewayStateData = GatewayStateData(data)
```

Comment (gslevinkov) [reply solanaj/src/main/java/org/p2p/solanaj/kits/renBridge/GatewayStateData.kt:17]: when should i use companion object, and when constructor?

File: solanaj/src/main/java/org/p2p/solanaj/rpc/RpcSolanaInteractor.kt [hunk 6]
```diff
@@ -52,9 +53,9 @@ class RpcSolanaInteractor(
             throw IllegalStateException("Chain not initialized")
         }
         val sHash = Base58.encode(Hash.generateSHash())
-        val index = this.gatewayRegistryData?.selectors?.indexOf(sHash) ?: -1
+        val index = this.gatewayRegistryData?.selectors?.indexOf(sHash)
 
-        return gatewayRegistryData?.gateways?.get(index)
+        return index?.let { gatewayRegistryData?.gateways?.get(it) }
             ?: throw IllegalStateException("Gateway not found at index $index")
```
Comment (gslevinkov) [inline solanaj/src/main/java/org/p2p/solanaj/rpc/RpcSolanaInteractor.kt:59]:
```suggestion
        return index?.let { gatewayRegistryData?.gateways?.get(it) } ?: error("Gateway not found at index $index")
```

**Review by gslevinkov — APPROVED** [review]

File: solanaj/src/main/java/org/p2p/solanaj/kits/renBridge/GatewayStateData.kt (hunk 5 above)
Comment (davran1001) [reply solanaj/src/main/java/org/p2p/solanaj/kits/renBridge/GatewayStateData.kt:17]: I don't know, i just move it from nested class

Comment (davran1001) [reply solanaj/src/main/java/org/p2p/solanaj/kits/renBridge/GatewayStateData.kt:17]: you can use any

**Pull request merged**